components/shared/net/request.rs

```diff
@@ -728,6 +729,40 @@ pub fn is_cors_safelisted_method(m: &Method) -> bool {
     matches!(*m, Method::GET | Method::HEAD | Method::POST)
 }
 
+/// <https://fetch.spec.whatwg.org/#cors-safelisted-request-header>
+pub fn is_cors_safelisted_request_range(value: &[u8]) -> bool {
```

Comment:
You may want to add some unit tests for this code in https://github.com/servo/servo/tree/main/components/shared/net/tests, which should make it easier to track down if the WPT test is still failing because of parsing reasons or something else.

Reply:
You can run those unit tests with `./mach test-unit -p net_traits`.

Reply:
Thank you for the update. I see that the tests in cors-safelisted-request-header.any.html are currently failing, and that the expected outcomes for the preflight checks aren't being met for the given range values. I understand the importance of narrowing down whether the issue is due to parsing or another part of the process.

I will add some unit tests for the is_cors_safelisted_request_range function in components/shared/net/tests. This should help in isolating the issue and tracking down if it's related to the parsing logic or something else.

Once the unit tests are in place, I'll run them using ./mach test-unit -p net_traits to ensure everything works as expected.

Reply:
I’ve added unit tests for CORS safelisted headers to ensure the functionality is thoroughly covered. Here's a screenshot showing all tests passing successfully.
![Screenshot 2024-12-28 174815](https://github.com/user-attachments/assets/b741eecf-ceeb-4b28-962a-3a907562ac6b)
Please let me know if there’s anything you’d like to adjust or expand on. Thank you!